### ISSUE TITLE
Matching mocap and robot coordinate systems

### ISSUE THREAD

**Issue by harpreets** [CreativeInquiry/ofxRobotArm#5]
How is this being done in the mocap examples? The documentation says "If you place a rigid body on the robot's TCP, you should see it moving & reorienting in the app the same as in real life." 

In my case, the streamed rigid body position is quite far from the robot in the simulation even though it is touching the tool connector in the physical world.

Is this supposed to be done through Motive by adjusting pivot positions? Or specifying certain position/orientation for the robot in the application? 
It'd be great if this could be unwrapped for details so that I can carry out similar alignment routine (for position/orientation) as well.

**Comment by madelinegannon** [comment]
Yes, let me clarify... 

It's best to align the motion capture coordinate system to the robot's coordinate system (and not the other way around).  For our setup, we printed an adapter to for the robot to hold the Optitrack's calibration base –– that way we always know where the base is in relation to the robot.

You'll need to do a few transformations (an XYZ offset and 1 or 2 rotations) depending on your robot's position when you're calibrating to mocap world. The way we check if our coordinate systems are aligned after doing our wanding is to:

1. Attach the Optitrack calibration base to the end of the robot, and Set Base in Motive.
2. Rotate axes in Motive to match axes of robot
3. Offset origin in Motive to match the base of the robot
4. Stream calibration base as a rigid body to `mocap-example`
5. Visually inspect if the markers are in the right place; if the virtual rigid body is not near the TCP of the simulation robot, then you know a rotation in Step 2 or an offset in Step 3 is off.
6. Move the robot in the X+/X- direction and visually verify the rigid body in `mocap-example` is moving in the same direction
7. Repeat for the robot's Y and Z axis; if the virtual rigid body is moving in a different direction actual rigid body on the robot, then you know a rotation in Step 2 is off.

The horizontal edge of the calibration base is +X and the vertical edge is +Z (UP is +Y by default in Motive, but we set up our data streaming to use Z as up to make the transformations easier).
